scylla/src/transport/load_balancing/default.rs

```diff
@@ -323,11 +330,18 @@ impl DefaultPolicy {
         Self::randomly_rotated_nodes(nodes).filter(predicate)
     }
 
-    fn shuffle<'a>(iter: impl Iterator<Item = NodeRef<'a>>) -> impl Iterator<Item = NodeRef<'a>> {
+    fn shuffle<'a>(
+        &self,
+        iter: impl Iterator<Item = NodeRef<'a>>,
+    ) -> impl Iterator<Item = NodeRef<'a>> {
         let mut vec: Vec<NodeRef<'a>> = iter.collect();
 
-        let mut rng = thread_rng();
-        vec.shuffle(&mut rng);
+        if let Some(fixed) = self.fixed_shuffle_seed {
+            let mut gen = Pcg32::new(fixed as u64, 0);
+            vec.shuffle(&mut gen);
```

Comment:
Is the cost of calling `Pcg32::new()` on each query negligible, or should we take it into account?

Reply:
It should be pretty cheap: https://docs.rs/rand_pcg/latest/src/rand_pcg/pcg64.rs.html#88-92. I can see one multiplication, two additions and some operations on bits.